slack_sdk/models/blocks/basic_components.py

```diff
@@ -249,7 +249,9 @@ def to_dict(self, option_type: str = "block") -> dict:  # skipcq: PYL-W0221
         self.validate_json()
         if option_type == "dialog":  # skipcq: PYL-R1705
             return {"label": self.label, "value": self.value}
-        elif option_type == "action":
+        elif option_type == "action" or option_type == "attachment":
+            # "action" can be confusing but it means a legacy message action in attachments
```

Comment:
I will work on better comments and documentation in a pull request for  #992 later